### PR TITLE
upgrade to 0.6

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

File: src/BinDeps.jl [hunk 1]
```diff
@@ -6,8 +6,8 @@ using Compat
 
 export @make_run, @build_steps, find_library, download_cmd, unpack_cmd,
     Choice, Choices, CCompile, FileDownloader, FileRule,
-    ChangeDirectory, FileDownloader, FileUnpacker, prepare_src,
```
Comment (KristofferC) [inline src/BinDeps.jl]: The removed ones does not exist in the package.

File: src/BinDeps.jl [hunk 2]
```diff
@@ -297,8 +297,6 @@ macro dependent_steps(steps)
     blk
 end
 
-mypwd() = chomp(readall(`pwd`))
```
Comment (KristofferC) [inline src/BinDeps.jl]: Not used, and `readall` doesn't exist.

**Comment by KristofferC** [comment]
bump @ararslan

**Comment by ararslan** [comment]
Sorry, I didn't see this. I guess I'm not following BinDeps on GitHub.

See #319 and #314. As much as I'd like to, I guess we can't drop 0.5 yet. Tony mentioned that we may be able to do 0.7-compatible releases from a separate branch.

File: src/BinDeps.jl (hunk 2 above)
Comment (ararslan) [reply src/BinDeps.jl]: Hah, that's also equivalent to the Julia function `pwd()`.

File: src/BinDeps.jl (hunk 1 above)
Comment (ararslan) [reply src/BinDeps.jl]: ?? I'm pretty sure `ChangeDirectory` does, as I think I have a package that uses it.

**Comment by KristofferC** [comment]
Just do backports to a 0.5 branch if needed? That's how Documenter does it.

**Comment by ararslan** [comment]
Yes, that's what I'd prefer to do. Any objections, @tkelman?

**Comment by ararslan** [comment]
CI failures are unrelated. Travis on Linux 0.7 is an error fetching Julia and 32-bit Windows on AppVeyor is due to WinRPM.

**Comment by musm** [comment]
could we get a tag with these changes?

**Comment by tkelman** [comment]
No, BinDeps master is not safe to tag from right now.

**Comment by KristofferC** [comment]
What does this mean?

**Comment by ararslan** [comment]
Tony explained in https://github.com/JuliaLang/METADATA.jl/pull/11896. There are a couple of PRs we need to get in first.

**Pull request merged**